### DISCUSSION TITLE
Flood + qbt: Torrent Management

### DISCUSSION THREAD

**Discussion by rg9400** [jesec/flood#209] [Q&A]
I recently installed a container that packages qbittorrent + flood together, and I've been really liking the interface. I've been trying to see how to best leverage Flood, but I've been struggling with a few things that I thought were worth asking about.

1. When I set qbt's Default Torrent Management Mode to Automatic, Flood will always download to the default save path. I tried setting the download directory in Radarr and even in the native Flood RSS feed, but everytime a torrent is added, Flood ignores these paths and uses the default path from qbt. I assumed it is related to not having a Category in qbt (vs a tag), so I switched the Torrent Management Mode to Manual. However, this poses a problem as not everything I use integrates natively with Flood (LazyLibrarian, Mylar, etc.). For those apps, I use qbt Categories with predefined paths. With a manual mode, anything added to qbt directly, even with a specified category, ignores the paths and downloads to the default download location. This is causing issues as I would like to leverage Flood, but I have very different download paths based on the application.

Long story short, is there a way to configure the settings in qbt/Flood such that applications can interact with either frontend without having to download to the default location?

2. In the Arrs, there is a note that says they cannot remove completed downloads from Flood, though such functionality exists natively in qbt. Is this a limitation at Flood's end or the Arr integration? Is there a benefit to running Flood as the download client in the Arrs vs just running qbt directly, since that integration seems to allow for Completed Download Handling?

**Comment by jesec** [comment]
I have absolutely no intention to support qBittorrent's category and associated automatic torrent management (ATM), which is inconsistent and opaque. It is also unnecessary and confusing to have two taxonomy systems. Plus, upstream qBittorrent is evaluating removal of ATM in their PR 13773. 

It is much more consistent to specify destination path directly in downstream integrations, and you should do that.

`Remove completed downloads` in *arr means removal after certain criteria (e.g. seed ratio, seed time) are met. It is not unconditional. Flood does not support seed ratio/time management yet, so *arr is not going to remove torrents. Otherwise, it would be "instant removal/hit-and-run", which is not in the best interests of BitTorrent community. While it is possible to manage seed ratio via *arr, as implemented in the original Flood support commit, Sonarr developers are not comfortable with the idea (https://github.com/Sonarr/Sonarr/pull/4159#discussion_r551041337).
